__test__/users.test.js

```diff
@@ -272,3 +275,220 @@ describe('POST user update social media info', () => {
+        expect(res.body.message).toBe(
+          'User already registered with username or email provided'
+        );
+      });
+  });
+  it('should return a 201 when user is created succesful', async () => {
+    const user = {
+      username: 'Johnson',
+      email: 'gmail@gmail.com',
+      password: '123456789'
+    };
+    return request(server)
+      .post('/api/user/signup')
+      .send(user)
+      .then(res => {
+        expect(res.status).toBe(201);
+        expect(res.body.data.user.email).toBe(user.email);
+      });
+  });
+});
+
+describe('POST Login User', () => {
+  it('should return 400 if email type is not valid', () => {
+    const user = {
+      email: 'email'
+    };
+    return request(server)
+      .post('/api/user/login')
+      .send(user)
+      .then(res => {
+        expect(res.status).toBe(400);
+        expect(res.body.message).toBe('Input a valid email');
+      });
+  });
+  it('should return 404 if user doesn exists', () => {
+    const user = {
+      email: 'email@gmail.com'
+    };
+    return request(server)
+      .post('/api/user/login')
+      .send(user)
+      .then(res => {
+        expect(res.status).toBe(404);
+        expect(res.body.message).toBe('User not found');
+      });
+  });
+  it('should return 401 if credientials are not valid', () => {
+    const user = {
+      email: 'nmereginivincent@gmail.com',
+      password: '123456789'
+    };
+    return request(server)
+      .post('/api/user/login')
+      .send(user)
+      .then(res => {
+        expect(res.status).toBe(401);
+        expect(res.body.message).toBe('Invalid credentials');
+      });
+  });
+  it('should return 200 when details are correct', () => {
+    const user = {
+      email: 'nmereginivincent@gmail.com',
+      password: 'password'
+    };
+    return request(server)
+      .post('/api/user/login')
+      .send(user)
+      .then(res => {
+        expect(res.status).toBe(200);
+        expect(res.body.data.message).toBe(
+          `${user.email} successfully logged in.`
+        );
+      });
+  });
+});
+
+describe('GET /auth/github', () => {
+  it('should return 302', () => {
+    jest
+      .spyOn(authMiddleware, 'githubStrategy')
+      .mockResolvedValue({ success: true });
+    return request(server)
+      .get('/api/user/auth/github/')
+      .then(res => {
+        expect(res.status).toBe(302);
+      });
+  });
+  it('should return 302', () => {
+    jest
+      .spyOn(authMiddleware, 'githubStrategy')
+      .mockResolvedValue({ success: true });
+    jest.spyOn(passport, 'authenticate').mockResolvedValue({ success: true });
+    return request(server)
+      .get('/api/user/auth/github/callback')
+      .then(res => {
+        expect(res.status).toBe(302);
+      });
+  });
+});
+
+describe('PATCH /api/user/:username/password', () => {
+  it('should return 401 if no token is provided', () => {
+    return request(server)
+      .patch('/api/user/:vincent/password')
```

Comment:
Do we really need the `:` before the username here?

Reply:
THIS testing for invalid username